### PR TITLE
Feature create move

### PR DESCRIPTION
**Pull request by Kathalania**

### CODE REVIEW

File: backend/src/test/java/de/neuefische/backend/MoveIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class MoveIntegrationTest {
+
+    @Autowired
+    MockMvc mockMvc;
+    @Autowired
+    MoveInterface moveInterface;
```
Comment (FullOfPit) [inline backend/src/test/java/de/neuefische/backend/MoveIntegrationTest.java:27]: You should not be mocking your repository in integration tests

File: backend/src/test/java/de/neuefische/backend/MoveServiceUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+class MoveServiceUnitTest {
+    @Test
+    void addMove_ShouldReturnAddedMove_WhenMoveAdded(){
+        //GIVEN
+        final MoveInterface moveInterface = mock(MoveInterface.class);
+        final MoveService moveService = new MoveService(moveInterface);
+
```
Comment (FullOfPit) [inline backend/src/test/java/de/neuefische/backend/MoveServiceUnitTest.java:17]: Why have you made these mocked classes final?

File: backend/src/test/java/de/neuefische/backend/MoveServiceUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                "6-count", "geschlossen", "offen");
+        when(moveInterface.save(tuckTurn))
+                .thenReturn(tuckTurn);
+        //WHEN
+        Move actual = moveService.addMove(tuckTurn);
+
+        //THEN
+        verify(moveInterface).save(tuckTurn);
+        assertEquals(actual, tuckTurn);
+
```
Comment (FullOfPit) [inline backend/src/test/java/de/neuefische/backend/MoveServiceUnitTest.java:28]: What happens when you try to add a move that does not contain all fields?

File: frontend/src/AddMove.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            count: count,
+            start: start,
+            end: end
+        }
+        props.addMove(newMove)
+        navigate("/moves")
+    }
+
+    return (
+        <div>
```
Comment (FullOfPit) [inline frontend/src/AddMove.tsx:34]: What's the additional div for?

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -1,29 +1,26 @@
...
-  );
+
+    const {addMove} = useMoves()
+
+    return (
+        <div className="App">
+            <BrowserRouter>
+                <Header/>
+                <Routes>
+                    <Route path="/add"
```
Comment (FullOfPit) [inline frontend/src/App.tsx:17]: Ambiguous name for a path, would change that to something more descriptive

File: frontend/src/Move.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+export type NewMove = {
+    name: string,
+    description: string,
+    style: string,
+    count: string,
+    start: string,
+    end: string
+}
```
Comment (FullOfPit) [inline frontend/src/Move.ts:18]: Why have two explicit types that are highly similar? NewMove could be used a Move if id was an optional field (hint: "id?")

File: frontend/src/NavigationBar.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import { useLocation, useNavigate } from "react-router-dom";
+import "./NavigationBar.css";
+
+export default function NavigationBar() {
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const [value, setValue] = useState(location.pathname);
+
+    const handleChange = (event: any, newValue: string) => {
```
Comment (FullOfPit) [inline frontend/src/NavigationBar.tsx:12]: It's not good to type event as any; could you change that?

File: frontend/src/useMoves.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import {Move, NewMove} from "./Move";
+import axios from "axios";
+import {useState} from "react";
+
+export default function useMoves() {
+
+    const [move,setMove] = useState('')
```
Comment (FullOfPit) [inline frontend/src/useMoves.ts:7]: Would set the type for this state and use an explicitly set move object as initial state

**Review by FullOfPit — COMMENTED** [review]
Just a few minor comments

**Review by CodingRobIT — APPROVED** [review]

**Pull request merged**